### PR TITLE
Backend/bug fix/#1530/money to high prec money changes

### PR DESCRIPTION
**Pull request by DwivediAman-exe**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Shared Kernel PR : https://github.com/nammayatri/shared-kernel/pull/178

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/mocks/public-transport-provider-platform/src/MockData/OnSearch.hs [hunk 1]
```diff
@@ -124,7 +124,7 @@
 
 buildFare :: Text -> Text -> HighPrecMoney -> Fare
 buildFare id route_id amount = do
-  let price = Price "INR" $ realToFrac amount
+  let price = Price "INR" $ amount
```
[the remark this answers is not in the document]
Comment (0utkarsh) [reply Backend/app/mocks/public-transport-provider-platform/src/MockData/OnSearch.hs:127]: Can you resolve these comments?

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/DriverFeeUpdates/DriverFee.hs [hunk 2]
```diff
@@ -107,9 +107,9 @@ unsubscribeDriverForPaymentOverdue Job {id, jobInfo} = withLogTag ("JobId-" <> i
         logInfo "Driver Not found. This should not be possible."
         throwError (InternalError "Driver Not Found") -- Unreachable
       Just driver -> do
-        let pendingAmount = driverFee.govtCharges + driverFee.platformFee.fee + round driverFee.platformFee.cgst + round driverFee.platformFee.sgst
+        let pendingAmount = driverFee.govtCharges + driverFee.platformFee.fee + driverFee.platformFee.cgst + driverFee.platformFee.sgst
         let paymentTitle = "Blocked - Payment Overdue"
-            paymentMessage = "Sorry, you are blocked from taking rides as payment of " <> show pendingAmount.getMoney <> " is pending for " <> show driverFee.numRides <> " ride(s) with a deadline of " <> show driverFee.payBy <> ". Please pay the balance amount to get unblocked."
+            paymentMessage = "Sorry, you are blocked from taking rides as payment of " <> show pendingAmount <> " is pending for " <> show driverFee.numRides <> " ride(s) with a deadline of " <> show driverFee.payBy <> ". Please pay the balance amount to get unblocked."
```
Comment (0utkarsh) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/DriverFeeUpdates/DriverFee.hs:112]: These are removed, can you rebase?

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Payment.hs [hunk 3]
```diff
@@ -68,11 +68,11 @@ createOrder (driverId, merchantId) driverFeeId = do
   driverPhone <- driver.mobileNumber & fromMaybeM (PersonFieldNotPresent "mobileNumber") >>= decrypt
   let driverEmail = fromMaybe "test@juspay.in" driver.email
   pendingFees <- QDF.findPendingFeesByDriverFeeId driverFee.id >>= fromMaybeM (DriverFeeNotFound $ getId driverFeeId)
-  let pendingAmount = fromIntegral pendingFees.govtCharges + fromIntegral pendingFees.platformFee.fee + pendingFees.platformFee.cgst + pendingFees.platformFee.sgst
+  let pendingAmount = pendingFees.govtCharges + pendingFees.platformFee.fee + pendingFees.platformFee.cgst + pendingFees.platformFee.sgst
   let createOrderReq =
         Payment.CreateOrderReq
           { orderShortId = driverFee.shortId.getShortId,
-            amount = round pendingAmount,
+            amount = pendingAmount,
```
Comment (0utkarsh) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Payment.hs:75]: I don't think this should be changed

**Review by 0utkarsh — CHANGES_REQUESTED** [review]

File: Backend/app/mocks/public-transport-provider-platform/src/MockData/OnSearch.hs (hunk 1 above)
[the remark this answers is not in the document]
Comment (DwivediAman-exe) [reply Backend/app/mocks/public-transport-provider-platform/src/MockData/OnSearch.hs]: Done

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/DriverFeeUpdates/DriverFee.hs (hunk 2 above)
Comment (DwivediAman-exe) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/DriverFeeUpdates/DriverFee.hs]: Done

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Payment.hs (hunk 3 above)
Comment (DwivediAman-exe) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Payment.hs]: Removed

**Pull request closed without merging**